sdk/containerregistry/container-registry/src/blob/models.ts

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { OperationOptions } from "@azure/core-client";
+
+/**
+ * Options for configuring the upload manifest operation.
+ */
+export interface UploadManifestOptions extends OperationOptions {
+  /**
+   * Tag to give the uploaded manifest
+   */
+  tag: string;
+}
+
+/**
+ * The result from uploading a blob to the registry.
+ */
+export interface UploadBlobResult {
+  /**
+   * The digest of the uploaded blob.
+   */
+  digest: string;
+}
+
+/**
+ * The result from downloading a manifest from the registry.
+ */
+export interface DownloadManifestResult {
+  /**
+   * The digest of the downloaded manifest as calculated by the registry.
+   */
+  digest: string;
+
+  /**
+   * The OCI manifest that was downloaded.
+   */
+  manifest: OciManifest;
+
+  /**
+   * The manifest stream that was downloaded.
+   */
+  manifestStream: NodeJS.ReadableStream;
+}
+
+/**
+ * The result from downloading a blob from the registry.
+ */
+export interface DownloadBlobResult {
+  /**
+   * The blob's digest, calculated by the resgistry.
+   */
+  digest: string;
+
+  /**
+   * The blob content.
+   */
+  content: NodeJS.ReadableStream;
+}
+
+/**
+ * The result from uploading a manifest to the registry.
+ */
+export interface UploadManifestResult {
+  /**
+   * The digest of the uploaded manifest, calculated by the registry.
+   */
+  digest: string;
+}
+
+/** Docker V2 image layer descriptor including config and layers. */
+export interface OciBlobDescriptor {
+  /** Layer media type */
+  mediaType?: string;
+  /** Layer size */
+  size?: number;
+  /** Layer digest */
+  digest?: string;
+  /** Specifies a list of URIs from which this object may be downloaded. */
+  urls?: string[];
+  /** Additional information provided through arbitrary metadata. */
+  annotations?: OciAnnotations;
+}
+
+/** Returns the requested OCI Manifest file. */
+export interface OciManifest {
+  /** Schema version */
+  schemaVersion: number;
+  /** V2 image config descriptor */
+  config?: OciBlobDescriptor;
+  /** List of V2 image layer information */
+  layers?: OciBlobDescriptor[];
+  /** Additional information provided through arbitrary metadata. */
+  annotations?: OciAnnotations;
+}
+
+/** Additional information provided through arbitrary metadata */
+export interface OciAnnotations {
+  /** Date and time on which the image was built (string, date-time as defined by https://tools.ietf.org/html/rfc3339#section-5.6) */
+  createdOn?: Date;
```

Comment:
Same question about optional. If we are sure some properties always come back then we can make it more accurate to help customer.

Reply:
These ones really are all optional ([spec](https://github.com/opencontainers/image-spec/blob/main/annotations.md)).